### PR TITLE
Added basic support for Azure Open AI

### PR DESCRIPTION
**Pull request by oorosh**
Added new LLM support, Azure Open AI. 

For accessing azure open ai api Azure.AI.OpenAI nuget package is used. This is official Microsoft package , currently in beta.

Updated versions Nuget packages.

### CODE REVIEW

File: src/LangChain.NET/LLMS/AzureOpenAi/AzureOpenAi.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        public override string ModelType { get; set; }
+        public override string LlmType { get; set; }
+        public override TikTokenizer Tokenizer { get; set; }
+
+        public override async Task<LlmResult> Generate(string[] prompts, List<string>? stopSequences)
+        {
+            var choices = new List<Choice>();
+            var usage = new List<CompletionsUsage>();
+
+            var client = new OpenAIClient(new Uri(_configuration.Endpoint), new AzureKeyCredential(_configuration.ApiKey));
```
Comment (CrazyBaran) [inline src/LangChain.NET/LLMS/AzureOpenAi/AzureOpenAi.cs]: Consider move OpenAIClient creation to constructor as one instance of BaseLlm can be use in multiple Chains, so we didnt create new instance for each call.

**Review by CrazyBaran — CHANGES_REQUESTED** [review]

File: src/LangChain.NET/LLMS/AzureOpenAi/AzureOpenAi.cs (hunk 1 above)
Comment (oorosh) [reply src/LangChain.NET/LLMS/AzureOpenAi/AzureOpenAi.cs]: fixed

File: tests/LangChain.NET.IntegrationTests/AzureOpenAiTests.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+            var chain = new LlmChain(new LlmChainInput(llm, prompt));
+
+            var result = await chain.Call(new ChainValues(new Dictionary<string, object>(1)
+        {
+            { "product", "colourful socks" }
+        }));
+
+            // The result is an object with a `text` property.
+            result.Value["text"].ToString().Should().NotBeEmpty();
```
Comment (CrazyBaran) [inline tests/LangChain.NET.IntegrationTests/AzureOpenAiTests.cs]: it should be string already there, so try `Value["Text"] as string`

**Review by CrazyBaran — CHANGES_REQUESTED** [review]

File: tests/LangChain.NET.IntegrationTests/AzureOpenAiTests.cs (hunk 2 above)
Comment (oorosh) [reply tests/LangChain.NET.IntegrationTests/AzureOpenAiTests.cs]: Fixed

Comment (CrazyBaran) [reply tests/LangChain.NET.IntegrationTests/AzureOpenAiTests.cs]: I think you didnt push a change :)

Comment (oorosh) [reply tests/LangChain.NET.IntegrationTests/AzureOpenAiTests.cs]: yes, forgot to push :D

**Review by jeastham1993 — APPROVED** [review]

**Comment by jeastham1993** [comment]
Happy to merge this, I've raised an RFC about the project structure though. I'm wondering if we should split each LLM implementation into its own library to make dependency management easier. Comments appreciated.

https://github.com/jeastham1993/langchain-dotnet/issues/4

**Pull request merged**